From d23018d276dfae889cc137d0cbd7f7c03506b3bd Mon Sep 17 00:00:00 2001
From: Wu Hao <hao.wu@intel.com>
Date: Thu, 7 May 2020 19:30:57 +0800
Subject: [PATCH] fpga: dfl: afu: add FOLL_LONGTERM flag when pin user pages

This patch add FOLL_LONGTERM flag in case the buffers are
used for DMA.

Signed-off-by: Wu Hao <hao.wu@intel.com>
Signed-off-by: Xu Yilun <yilun.xu@intel.com>
---
 drivers/fpga/dfl-afu-dma-region.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/fpga/dfl-afu-dma-region.c b/drivers/fpga/dfl-afu-dma-region.c
index b3e2abb195a0..4bf9aa148222 100644
--- a/drivers/fpga/dfl-afu-dma-region.c
+++ b/drivers/fpga/dfl-afu-dma-region.c
@@ -37,7 +37,7 @@ static int afu_dma_pin_pages(struct dfl_feature_dev_data *fdata,
 {
 	int npages = PFN_DOWN(region->length);
 	struct device *dev = &fdata->dev->dev;
-	unsigned int flags = 0;
+	unsigned int flags = FOLL_LONGTERM;
 	int ret, pinned;
 
 	ret = account_locked_vm(current->mm, npages, true);
